clang/lib/CIR/CodeGen/CIRGenExprComplex.cpp

```diff
@@ -0,0 +1,473 @@
+#include "CIRGenBuilder.h"
+#include "CIRGenCstEmitter.h"
+#include "CIRGenFunction.h"
+#include "clang/CIR/MissingFeatures.h"
+
+#include "mlir/IR/Value.h"
+#include "clang/AST/StmtVisitor.h"
+#include "llvm/Support/ErrorHandling.h"
+
+using namespace cir;
+using namespace clang;
+
+namespace {
+
+class ComplexExprEmitter : public StmtVisitor<ComplexExprEmitter, mlir::Value> {
+  CIRGenFunction &CGF;
+  CIRGenBuilderTy &Builder;
+  bool FPHasBeenPromoted;
+
+public:
+  explicit ComplexExprEmitter(CIRGenFunction &cgf)
+      : CGF(cgf), Builder(cgf.getBuilder()), FPHasBeenPromoted(false) {}
+
+  //===--------------------------------------------------------------------===//
+  //                               Utilities
+  //===--------------------------------------------------------------------===//
+
+  /// EmitLoadOfLValue - Given an expression with complex type that represents a
```

Comment:
Same as last comment, please remove these extra method names everywhere you have them on comments.

Reply:
Updated.